src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinInetProxyHelper.cs

```diff
@@ -178,5 +181,44 @@ public bool ManualSettingsOnly
 
             return useProxy;
         }
+
+        public bool AutoDetectScriptSuccess()
+        {
+            // Only check WinHttpDetectAutoProxyConfigUrl if auto detect is set.
+            if (!AutoDetect) return false;
+
+            bool doAutoDetect = false;
+            object executionLock = new object();
+            long currentTimeStamp = DateTime.Now.Ticks;
+
+            lock (executionLock)
+            {
+                doAutoDetect = _previousTimeStampTicks == DateTime.MinValue.Ticks ?
+                    true : (currentTimeStamp - _previousTimeStampTicks > TimerPeriod ? true : false);
+
+                _previousTimeStampTicks = currentTimeStamp;
+            }
```

Comment:
The lock isn't necessary.  If there's a race condition, worst case is more threads than we'd otherwise like will end up doing the auto-detect, and that's a lot better than adding global synchronization across all accesses.